javax.crypto.IllegalBlockSizeException when logging in

Found this repository through the article and thought it was great. While giving it a try I found a crash that might be worth noticing. 

Steps to reproduce:
1. Use biometrics to authenticate
2. On the `AuthorizedFragment` hit back 
3. Land in the "login" page again and the prompt opens up
4. Use Biometrics again to authenticate
5. See the Stack Trace below

Tested in both Android API 28 and 29

The reason is that we are using the same crypto object to authenticate twice because when you go back and re-subscribe to the `biometricParams` live data its sending the previous params

```
Process: com.redmadrobot.authenticateme, PID: 18327
    java.lang.Error: javax.crypto.IllegalBlockSizeException
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1173)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: javax.crypto.IllegalBlockSizeException
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:519)
        at javax.crypto.Cipher.doFinal(Cipher.java:2055)
        at com.redmadrobot.authenticateme.unauthorized_zone.login.pin.input.InputPinViewModel$authenticationCallback$1.onAuthenticationSucceeded(InputPinViewModel.kt:53)
        at androidx.biometric.BiometricFragment$2$2.run(BiometricFragment.java:138)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
     Caused by: android.security.KeyStoreException: Key user not authenticated
        at android.security.KeyStore.getKeyStoreException(KeyStore.java:1292)
        at android.security.keystore.KeyStoreCryptoOperationChunkedStreamer.update(KeyStoreCryptoOperationChunkedStreamer.java:132)
        at android.security.keystore.KeyStoreCryptoOperationChunkedStreamer.doFinal(KeyStoreCryptoOperationChunkedStreamer.java:217)
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:506)
```

--- a/app/src/main/java/com/redmadrobot/authenticateme/unauthorized_zone/login/pin/input/InputPinFragment.kt
+++ b/app/src/main/java/com/redmadrobot/authenticateme/unauthorized_zone/login/pin/input/InputPinFragment.kt
@@ -55,7 +55,9 @@ class InputPinFragment : Fragment() {
         })
 
         viewModel.biometricParams.observe(viewLifecycleOwner, Observer { params ->
-            showBiometricPrompt(params)
+            params.getContentIfNotHandled()?.let {
+                showBiometricPrompt(it)
+            }
         })
     }
 

--- a/app/src/main/java/com/redmadrobot/authenticateme/unauthorized_zone/login/pin/input/InputPinViewModel.kt
+++ b/app/src/main/java/com/redmadrobot/authenticateme/unauthorized_zone/login/pin/input/InputPinViewModel.kt
@@ -31,7 +31,7 @@ class InputPinViewModel(application: Application) : AndroidViewModel(application
 
     val biometricErrorMessage = MutableLiveData<SingleLiveEvent<String>>()
 
-    val biometricParams = MutableLiveData<BiometricParams>()
+    val biometricParams = MutableLiveData<SingleLiveEvent<BiometricParams>>()
 
     private val preferences by lazy { getApplication<App>().encryptedStorage }
     private val aead by lazy { getApplication<App>().pinSecuredAead }
@@ -105,7 +105,7 @@ class InputPinViewModel(application: Application) : AndroidViewModel(application
                     val promptInfo = createPromptInfo()
                     val cryptoObject = BiometricPrompt.CryptoObject(createCipher())
 
-                    biometricParams.value = BiometricParams(promptInfo, cryptoObject)
+                    biometricParams.value = SingleLiveEvent(BiometricParams(promptInfo, cryptoObject))
                 }
             }
         } else {

